envoy/config/rbac/v2alpha/rbac.proto

```diff
@@ -0,0 +1,155 @@
+syntax = "proto3";
+
+import "validate/validate.proto";
+
+package envoy.config.rbac.v2alpha;
+
+// RBAC (Role Based Access Control) defines ServiceRole and ServiceRoleBinding
+// objects.
+//
+// A ServiceRole specification includes a list of rules (permissions). Each rule has
+// the following standard fields:
+// * services: a list of services.
+// * verbs: HTTP verbs. In the case of gRPC, this field is ignored because the value is always
+// "POST".
+// * paths: HTTP paths or gRPC methods. Note that gRPC methods should be
+//   presented in the form of "packageName.serviceName/methodName".
+//
+// The fields used in ServiceRole definition map to attributes defined in AttributeContext
+// (//envoy/service/auth/v2/attribute_context.proto) as follows:
+//   service: destination.service
+//   method: request.http.method
+//   path: request.http.path
+//
+//  In addition, users can specify "constraints" on any attributes defined in AttributeContext
+//  (//envoy/service/auth/v2/attribute_context.proto).
+//
+// Below is an example of ServiceRole object "product-viewer", which has "read" ("GET" and "HEAD")
+// access to "products.default.svc.cluster.local" service at versions "v1" and "v2".
+//
+//     kind: ServiceRole
+//     metadata:
+//       name: products-viewer
+//       namespace: default
+//     spec:
+//       rules:
+//       - services: ["products.default.svc.cluster.local"]
+//         paths: ["/products/*"]
+//         verbs: ["GET", "HEAD"]
+//         constraints:
+//         - key: request.http.headers["version"]
+//           values: ["v1", "v2"]
+//
+// A ServiceRoleBinding specification includes two parts:
+// * "roleRef" refers to a ServiceRole object.
+// * A list of "subjects" that are assigned the roles.
+//
+// A subject can be a "user" or a "group", or represented with a set of attributes. Specifically,
```

Comment:
I think we need a really user friendly docs writeup of how this is all supposed to tie together. Thinking about this, and not knowing much about RBAC beyond the textbook parts, I'm guessing this all works because we can identify subjects via client certificates. How would this function without client certs? I think it would be helpful to explain overall context, provide an architecture diagram, explain how it works specifically for the kind of setup we have in places like Istio with mTLS, as well as how it can be adapted.

Reply:
+1 (my other comments)

Reply:
I added a README.md. Please take a look to see it answers all the questions here.